llvm/lib/Analysis/LazyValueInfo.cpp

```diff
@@ -645,7 +645,7 @@ LazyValueInfoImpl::solveBlockValueImpl(Value *Val, BasicBlock *BB) {
   // instruction is placed, even if it could legally be hoisted much higher.
   // That is unfortunate.
   PointerType *PT = dyn_cast<PointerType>(BBI->getType());
-  if (PT && isKnownNonZero(BBI, DL))
+  if (PT && isKnownNonZero(BBI, /*Depth=*/0, DL))
```

Comment:
Side note: We're currently inconsistent about this, but I think it would be best if we moved all the depth parameters after SQ, because end-users of the API don't need to specify depth.

Reply:
If you aren't going to implement that, I will after this patch lands.

Reply:
The fact that this PR changes it so that `isKnownNonZero(BBI, DL)` is valid in LLVM 18, used to be valid in LLVM 19, is currently an error in LLVM 19, but will become valid in LLVM 19 again is unfortunate and difficult to account for in downstream projects. @goldsteinn If you are going to change that, do you think it is worth updating downstream code to support the current API, or is that change going to go in fast enough that we can just wait?

Reply:
If I don't make the llvm19 cutoff, I'll drop it.

Reply:
Created #88873 for it.

Reply:
Thank you for making the PR :)